### PR TITLE
Add test for WD and FPWD regarding shortname

### PR DESCRIPTION
**Pull request by jennyliang220**
- Remove unused variable, [resolves](https://github.com/w3c/specberus/issues/1646)
- Add check for FPWD and WD, FPWD should use a unique shortname while WD uses an existing one, [resolves](https://github.com/w3c/specberus/issues/1644)
- add test to check shortname [TBD]

### CODE REVIEW

File: lib/l10n-en_GB.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ export const messages = {
         'Cannot find the link of the W3C profile in id="w3c-state" element pointing to "https://www.w3.org/standards/types#xx".',
     'headers.w3c-state.wrong-w3c-state-link':
         'The link for text "${text}" should pointing to W3C profile "${expectedLink}", but the link found in the document is "${linkFound}".',
+    'headers.w3c-state.shortname-existed':
+        'The ${status} should use an unique shortname".',
```
Comment (deniak) [inline lib/l10n-en_GB.js:27]:
```suggestion
        'The ${status} should use an unique shortname.',
```

File: lib/l10n-en_GB.js [hunk 2]
```diff
@@ -23,6 +23,10 @@ export const messages = {
...
+    'headers.w3c-state.shortname-not-existed':
+        'The ${status} should use an existing shortname".',
```
Comment (deniak) [inline lib/l10n-en_GB.js:29]:
```suggestion
        'The ${status} should use an existing shortname.',
```

**Review by deniak — CHANGES_REQUESTED** [review]

**Comment by jennyliang220** [comment]
Pubrules requires the shortname mentioned in **This Version** and **Supersades this Recommendation**, as well as **Obsoletes this Recommendation** to be the same. 

https://github.com/w3c/specberus/blob/bf01c4463b297c4a691a622b663da784b9321b2a/lib/l10n-en_GB.js#L95-L96

Some [test errors](https://github.com/w3c/specberus/actions/runs/3218424407/jobs/5262581561#step:6:2857) in this PR are related.

In case of FPWD, it's intended to replace the Rec document, so it should use a different shortname, @deniak WDYT?

Screen Shot 2022-10-10 at 5 13 13 PM

Screen Shot 2022-10-10 at 5 08 53 PM

**Comment by deniak** [comment]
> In case of FPWD, it's intended to replace the Rec document, so it should use a different shortname, @deniak WDYT?

Yes, shortnames should be different. However, I don't think I've ever seen that use case (FPWD +  Supersedes|Obsoletes this Recommendation). Especially now with the new process where groups have a way to update an existing Recommendation without too much hassle.

**Comment by jennyliang220** [comment]
> shortnames should be different. However, I don't think I've ever seen that use case (FPWD + Supersedes|Obsoletes this Recommendation).

Searching recent 700 documents back to 2014, there's no **Supersades this Recommendation** nor **Obsoletes this Recommendation** mentioned in the headers. Maybe we could drop the test, as well as the rule.

File: lib/validator.js [hunk 3]
```diff
@@ -754,23 +753,15 @@ Specberus.prototype.isFP = async function () {
...
+    this.isFirstPublic = !previousLink;
     return this.isFirstPublic;
 };
 
 // get previous version link from API using shortname
 Specberus.prototype.getPreviousVersion = async function () {
     const dts = this.extractHeaders();
     const shortname = this.shortname || (await this.getShortname());
-
+    if (!shortname) return false;
```
Comment (deniak) [inline lib/validator.js:764]:
```suggestion
    if (!shortname) return;
```

**Review by deniak — CHANGES_REQUESTED** [review]

File: lib/rules/headers/w3c-state.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ export const { name } = self;
  * @param sr
  * @param done
  */
-export function check(sr, done) {
+export async function check(sr, done) {
```
Comment (jennyliang220) [inline lib/rules/headers/w3c-state.js:17]:
```suggestion
export function check(sr, done) {
```

[the remark this answers is not in the document]
Comment (deniak) [reply lib/rules/headers/w3c-state.js]: reverted in https://github.com/w3c/specberus/pull/1648/commits/0f3d587755b716afb0ec0079a5aeb83a4f30179b

File: lib/rules/headers/dl.js [hunk 5]
```diff
@@ -382,5 +318,31 @@ export async function check(sr, done) {
...
         sr.error(editorError, 'editor-not-found');
     }
+
+    // check shortname is valid.
+    const isFP = await sr.isFP();
+    // FP documents cannot use existing shortname
+    if (
+        sr.config.longStatus === 'First Public Working Draft' &&
+        !isFP &&
+        shortname
```
[the remark this answers is not in the document]
Comment (jennyliang220) [reply lib/rules/headers/dl.js:328]: If shortname in headers not extracted (edge case exists in test), there's no previous version link, `isFP` is true, it'll go into the `if` condition. So `shortname` is necessary.

[the remark this answers is not in the document]
Comment (jennyliang220) [reply lib/rules/headers/dl.js:329]: // check shortname before send the request in getPreviousVersions()

**Comment by jennyliang220** [comment]
separate a new rule `shortname.js` to check shortname existence for REC

File: lib/l10n-en_GB.js [hunk 6]
```diff
@@ -53,10 +53,25 @@ export const messages = {
...
+    'headers.shortname.this-previous-shortname':
+        '<em>Shortnames</em> differ between This version (${thisShortname}) and Previous Versions (${previousShortname}).',
+    'headers.shortname.history-syntax':
+        'Wrong syntax for History link, please use the link "https://www.w3.org/standards/history/${shortname}".',
+    'headers.shortname.history-bad-previous':
+        'This document contains a shortname change with previous shortname "${previousShortname}", but the it seems be not correct because "${url}" doesn\'t exist.',
+    'headers.shortname.this-rescinds-shortname':
+        '<em>Shortnames</em> differ between This Version (${thisShortname}) and Rescinds this Recommendation (${rescindsShortname}).',
+    'headers.shortname.shortname-existed':
+        'The ${status} should use an unique shortname.',
```
Comment (deniak) [inline lib/l10n-en_GB.js:70]: The status will always be 'First Public Working Draft' the way it's being used in `shortname.js`. No need to pass an extra parameter.
```suggestion
        'FPWDs must use a new shortname.',
```

File: lib/rules/headers/shortname.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+    const isFP = await sr.isFP();
+    // FP documents cannot use existing shortname
+    if (
+        sr.config.longStatus === 'First Public Working Draft' &&
+        !isFP &&
+        shortname
+    ) {
+        sr.error(self, 'shortname-existed', {
+            status: sr.config.longStatus,
+        });
```
Comment (deniak) [inline lib/rules/headers/shortname.js:187]: changes related to the suggestion in `l10-en_GB.js`.
```suggestion
        sr.error(self, 'shortname-existed');
```

**Review by deniak — CHANGES_REQUESTED** [review]

File: lib/l10n-en_GB.js [hunk 8]
```diff
@@ -53,10 +53,25 @@ export const messages = {
...
     'headers.h1-title.not-match':
         "Content of title and h1 do not match. Text of title is '${titleText}' while h1 is (transformed into) '${h1Text}'.",
+    // headers/shortname
+    'headers.shortname.shortname-lowercase':
+        'The shortname of the document must in lowercase, but `${shortname}` is found.',
+    'headers.shortname.this-latest-shortname':
+        '<em>Shortnames</em> differ between This version (${thisShortname}) and Latest Versions (${latestShortname}).',
+    'headers.shortname.this-previous-shortname':
+        '<em>Shortnames</em> differ between This version (${thisShortname}) and Previous Versions (${previousShortname}).',
+    'headers.shortname.history-syntax':
```
[the remark this answers is not in the document]
Comment (jennyliang220) [reply lib/l10n-en_GB.js:63]: Yes, the existence of history link is checked [here in dl.js](https://github.com/w3c/specberus/pull/1648/files/eace9740d9068b45d3fe2ec77b1a38dd0803fa85#diff-cf098e75516357312f91fb778a977b2fb8e6473b3130ad33efb1ae48c36cf085R65), and the format is checked in [later in dl.js](https://github.com/w3c/specberus/pull/1648/files#diff-cf098e75516357312f91fb778a977b2fb8e6473b3130ad33efb1ae48c36cf085R134) , I move the shortname check to the new `shortname.js` rule.

[the remark this answers is not in the document]
Comment (deniak) [reply lib/l10n-en_GB.js:63]: Indeed!

File: lib/rules/headers/shortname.js [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+                    const historyHref = linkHistory.getAttribute('href');
+                    // Check if the history link exist
+                    let historyStatusCode;
+                    try {
+                        const res = await doAsync(superagent).head(historyHref);
+                        historyStatusCode = res.statusCode;
+                    } catch (err) {
+                        historyStatusCode = err.status;
+                    }
+                    var hasPreviousVersion = !(await sr.isFP());
```
[the remark this answers is not in the document]
Comment (jennyliang220) [reply lib/rules/headers/shortname.js:117]: `Specberus.prototype.isFP` saves the result of the Promise
